File: Module2/chapter/abstract-factory/iteration-2/src/paystation/domain/Receipt.java
package paystation.domain;

import java.io.PrintStream;

/** This interface models the receipt that the customer receives from
    the pay station.

    Author: (c) Henrik Bærbak Christensen 2006
*/

public interface Receipt {
  /** return the number of minutes this receipt is valid for.
   *  @return number of minutes parking time
   */
  int value();
  
  /** prints the receipt on the given PrintStream.
   */
  void show(PrintStream stream);
}

File: Module2/chapter/tdd/iteration-1-C#/Receipt.cs
/** This interface models a receipt defining the amount of 
    parking time bought.

    author: (c) Henrik Bærbak Christensen 2005
*/

public interface Receipt {
  /** return the number of minutes this receipt is valid for.
      @return number of minutes parking time
  */
  int value();
  
  /** "shows" the receipt by writing text to the 
      shell/prompt.
  */
  void show();
}

